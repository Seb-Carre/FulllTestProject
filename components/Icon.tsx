import {
  Image,
  ImageProps,
  ImageSourcePropType,
  ImageStyle,
  StyleProp,
} from 'react-native';

import React from 'react';

type IconParams = {
  name: ImageSourcePropType;
  style: StyleProp<ImageStyle>;
  props?: ImageProps;
};

/**
 * Functional Component to return a icon based on RN Image Component
 * You can't use dynamic variables to pass the icon file
 * See the doc about it https://reactnative.dev/docs/images#static-image-resources
 * Using a image from url is not a good pratice except if you have your own api ! better to have it in the project
 * @param {string} text
 */
export function Icon({name, style, props}: IconParams) {
  return <Image source={name} style={style} {...props} />;
}
