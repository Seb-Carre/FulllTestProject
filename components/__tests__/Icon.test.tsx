import {Icon} from '../Icon';
import React from 'react';
import {render} from '@testing-library/react-native';

const checkMark = '../../assets/images/check-mark.png';
it('render the icon as snapShot expected', () => {
  const {toJSON} = render(<Icon name={require(checkMark)} style={{flex: 1}} />);

  expect(toJSON()).toMatchSnapshot();
});

it('check if the image source is correct', () => {
  const {queryByTestId} = render(
    <Icon testID="icon" name={require(checkMark)} style={{flex: 1}} />,
  );
  // We need to add an extra path because Icon module is not in the same place
  expect(queryByTestId('icon')?.props.source).toStrictEqual({
    testUri: '../' + checkMark,
  });
});
