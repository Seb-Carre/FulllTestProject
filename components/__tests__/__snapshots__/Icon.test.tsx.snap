// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the icon as snapShot expected 1`] = `
<Image
  source={
    {
      "testUri": "../../../assets/images/check-mark.png",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
/>
`;
